src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -384,4 +384,39 @@ else if ( isDirectory )
         }
     }
 
+    // Visible for testing
+    protected boolean shouldExtractEntry( File file, String entryName, Date entryDate ) throws IOException
+    {
+        String canonicalDestPath = file.getCanonicalPath();
+        boolean fileOnDiskIsNewerThanEntry = ( file.lastModified() >= entryDate.getTime() );
+        boolean differentCasing = !StringUtils.equalsIgnoreCase( entryName, canonicalDestPath );
+
+        String casingMessage = String.format( "Archive entry '%s' and existing file '%s' names differ only by case."
+                + " This may cause issues on case-insensitive filesystems.", entryName, canonicalDestPath );
+
+        // Optimise for situation where we need no checks
+        if ( !file.exists() )
+        {
+            return true;
+        }
+
+        // (1)
+        if ( fileOnDiskIsNewerThanEntry )
+        {
+            // (3)
```

Comment:
Where is 2?

Reply:
Ah, that is combined with (4). Those are basically notes, referring to [Roberts table](https://github.com/codehaus-plexus/plexus-archiver/pull/128#discussion_r499219172). I think I should omit them, since the tests are more clear than those numbers. WDYT?

Reply:
Please add that table as internal. After the PR has been processed no one will remember this table.